build-system/tasks/lint.js

```diff
@@ -15,32 +15,46 @@
  */
 
 
-var gulp = require('gulp');
+var argv = require('minimist')(process.argv.slice(2));
+var config = require('../config');
 var eslint = require('gulp-eslint');
+var gulp = require('gulp');
+var lazypipe = require('lazypipe');
 var util = require('gulp-util');
-var config = require('../config');
+var watch = require('gulp-watch');
+
+var isWatching = (argv.watch || argv.w) || false;
 
 var options = {
   plugins: ['eslint-plugin-google-camelcase'],
 };
 
+var srcs = ['**/*.js', config.src.exclude];
+
+var watcher = lazypipe().pipe(watch, srcs);
+
 /**
  * Run the eslinter on the src javascript and log the output
  * @return {!Stream} Readable stream
  */
 function lint() {
   var errorsFound = false;
-  return gulp.src(['**/*.js', config.src.exclude])
-      .pipe(eslint(options))
-      .pipe(eslint.formatEach('compact', function(msg) {
-        errorsFound = true;
-        util.log(util.colors.red(msg));
-      }))
-      .on('end', function() {
-        if (errorsFound) {
-          process.exit(1);
-        }
-      });
+  var stream = gulp.src(srcs);
+
+  if (isWatching) {
+    stream = stream.pipe(watcher());
+  }
+
+  return stream.pipe(eslint(options))
+    .pipe(eslint.formatEach('compact', function(msg) {
+      errorsFound = true;
+      util.log(util.colors.red(msg));
+    }))
+    .on('end', function() {
+      if (errorsFound) {
+        process.exit(1);
```

Comment:
it does not actually. the moment the `gulp-watch` is attached to the stream it does not allow it to terminate (so never it never fires the `end` event).